testtime/testtime.go

```diff
@@ -0,0 +1,292 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package testtime provides a wrapper around time.Timer and its associated
+// functions so that timers can be mocked in tests.
+package testtime
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/snapcore/snapd/testutil"
+)
+
+// Timer is an interface which wraps time.Timer so that it may be mocked.
+//
+// Timer is fully compatible with time.Timer, except that since interfaces
+// cannot have instance variables, we must expose the C channel as a method.
+// Therefore, when replacing a time.Timer with testtime.Timer, any direct
+// accesses of C must be replaced with C().
+//
+// For more information about time.Timer, see: https://pkg.go.dev/time#Timer
+type Timer interface {
+	C() <-chan time.Time
+	Reset(d time.Duration) bool
+	Stop() bool
+}
+
+// useMockedTimers indicates whether AfterFunc and NewTimer should return
+// TestTimer instances instead of RealTimer instances.
+var useMockedTimers = false
+
+// MockTimers makes AfterFunc and NewTimer return TestTimers instead of
+// RealTimers. This can be reversed by calling the returned restore function.
+func MockTimers() (restore func()) {
```

Comment:
if the code always returns fake timer, then this is no longer needed, nor the other part checking use of MockTimers